portal/account/index_reset.php

```diff
@@ -0,0 +1,203 @@
+$valid = ((!empty(trim($_POST['uname']))) &&
+    (!empty(trim($_POST['login_uname']))) &&
+    (!empty(trim($_POST['pass_current']))) &&
+    (!empty(trim($_POST['pass_new']))) &&
+    (trim($_POST['uname']) == $auth[COL_POR_USER]) &&
+    (hash_equals(oemr_password_hash(trim($_POST['pass_current']), $auth[COL_POR_SALT]), $auth[COL_POR_PWD])));
+if (isset($_POST['submit'])) {
+    if (!$valid) {
+        $errmsg = xlt("Invalid Current Credentials Error.") . xlt("Unknown.");
+        $logit->portalLog('Credential update attempt', '', ($_POST['uname'] . ':unknown'), '', '0');
+        die($errmsg);
+    }
+    $code_current = trim($_POST['pass_current']);
+    $plain_code = trim($_POST['pass_new']);
+    $new_salt = oemr_password_salt();
+    $new_hash = oemr_password_hash($plain_code, $new_salt);
+    $sqlUpdatePwd = " UPDATE " . TBL_PAT_ACC_ON . " SET " . COL_POR_PWD . "=?, " . COL_POR_SALT . "=?, " . COL_POR_LOGINUSER . "=?" . " WHERE " . COL_ID . "=?";
+    privStatement($sqlUpdatePwd, array(
+        $new_hash,
+        $new_salt,
+        $_POST['login_uname'],
+        $auth[COL_ID]
+    ));
+}
+
+?>
+<!DOCTYPE html>
+<html>
+<head>
+    <title><?php echo xlt('Change Portal Credentials'); ?></title>
+    <?php
+    Header::setupHeader(['opener']);
+    if (!empty($_POST['submit'])) {
+        unset($_POST['submit']);
+        echo "<script>dlgclose();</script>\n";
+    }
+    ?>
+    <script type="text/javascript">
+        function checkUserName() {
+            let vacct = document.getElementById('uname').value;
+            let vsuname = document.getElementById('login_uname').value;
+            let data = {
+                'action': 'userIsUnique',
+                'account': vacct,
+                'loginUname': vsuname
+            };
+            $.ajax({
+                type: 'GET',
+                url: './account.php',
+                data: data
+            }).done(function (rtn) {
+                if (rtn === '1') {
+                    return true;
+                }
+                alert(<?php echo xlj('Log In Name is unavailable. Try again!'); ?>);
+                document.getElementById('login_uname').value = '';
+                document.getElementById('login_uname').focus();
+                return false;
+            });
+        }
+
+        function process_new_pass() {
+            if (document.getElementById('login_uname').value != document.getElementById('confirm_uname').value) {
+                alert(<?php echo xlj('The Username fields are not the same.'); ?>);
+                return false;
+            }
+            if (document.getElementById('pass_new').value != document.getElementById('pass_new_confirm').value) {
+                alert(<?php echo xlj('The new password fields are not the same.'); ?>);
+                return false;
+            }
+            if (document.getElementById('pass_new').value == document.getElementById('pass_new_confirm').value) {
```

Comment:
above line is incorrect. should have 'pass_current' somewhere in there.

Reply:
I guess now it is but I was not going to reverify a logged in password being changed. Not like openemr where more than one user has access to passwords changes for others. Here is only patient ever see this interface and they are already logged in. Still it was changed so nice catch.